cfgen/cfgen

```diff
@@ -462,10 +463,24 @@ def minion_id() -> Optional[str]:
         return f.readline().strip()
 
 
+def get_all_local_addrs():
+    interface_details = psutil.net_if_addrs()
+    ips = []
+    for iface in interface_details.values():
+        ips.append(iface[0].address)
+    return ips
+
+
 def is_localhost(hostname: str) -> bool:
     assert hostname
-    return hostname in ('localhost', '127.0.0.1', minion_id(),
-                        socket.gethostname(), socket.getfqdn())
+    m_id = minion_id()
+    if not m_id:
+        m_id = ""
+    for item in socket.gethostbyname_ex(hostname):
+        if any(x in item for x in ('localhost', '127.0.0.1', m_id,
+                                   socket.gethostname(), socket.getfqdn())):
+            return True
+    return socket.gethostbyname(hostname) in get_all_local_addrs()
```

Comment:
Ah sorry I missed this in my last comment, with above change (using gethostbyname_ex()) we don't need,
`return socket.gethostbyname(hostname) in get_all_local_addrs()` instead we can do,
```
for item in socket.gethostbyname_ex(hostname):
        if any(x in item for x in ('localhost', '127.0.0.1', m_id,
                                   socket.gethostname(), socket.getfqdn(), get_all_local_addrs())):
            return True
return False
```

Reply:
Okay, :) Also I think we dont need both localhost and 127.0.0.1. Either of them will be ok.
And we don't need to check socket.gethostname(), socket.getfqdn() as well. Since ` get_all_local_addrs()` will resolve all the local IP addresses.

Reply:
Another thing, Please correct me.
```
for item in socket.gethostbyname_ex(hostname):
        if any(x in item for x in (m_id, get_all_local_addrs())):
            return True
```
here `get_all_local_addrs()` is returning list. 
so `x` will be a `list` and `item` can be  `string or list`. (tuple of hostname, [alias], [ips])
So, `list in str` will be incorrect check right?

Reply:
So you can do, 
```
ids = [m_ids]
ids.extend(get_all_local_addrs())
for item in socket.gethostbyname_ex(hostname):
        if any(x in item for x in ids):
            return True
```
okay?

Reply:
Done, thanks!!